ports/stm/boards/nucleo_f746zg/board.dts

```diff
@@ -0,0 +1,933 @@
+			interrupt-names = "otghs", "ep1_out", "ep1_in";
+			num-bidir-endpoints = < 0x9 >;
+			ram-size = < 0x1000 >;
+			maximum-speed = "full-speed";
+			clocks = < &rcc 0x0 0x20000000 >;
+			phys = < &otghs_fs_phy >;
+			status = "disabled";
+			label = "OTGHS";
+		};
+		rtc: rtc@40002800 {
+			compatible = "st,stm32-rtc";
+			reg = < 0x40002800 0x300 >;
+			interrupts = < 0x29 0x0 >;
+			clocks = < &rcc 0x2 0x10000000 >;
+			prescaler = < 0x8000 >;
+			status = "okay";
+			label = "RTC_0";
+		};
+		adc1: adc@40012000 {
+			compatible = "st,stm32-adc";
+			reg = < 0x40012000 0x400 >;
+			clocks = < &rcc 0x3 0x100 >;
+			interrupts = < 0x12 0x0 >;
+			status = "okay";
+			label = "ADC_1";
+			#io-channel-cells = < 0x1 >;
+		};
+		dma1: dma@40026000 {
+			compatible = "st,stm32-dma";
+			#dma-cells = < 0x4 >;
+			reg = < 0x40026000 0x400 >;
+			interrupts = < 0xb 0x0 0xc 0x0 0xd 0x0 0xe 0x0 0xf 0x0 0x10 0x0 0x11 0x0 0x2f 0x0 >;
+			clocks = < &rcc 0x0 0x200000 >;
+			status = "disabled";
+			label = "DMA_1";
+		};
+		dma2: dma@40026400 {
+			compatible = "st,stm32-dma";
+			#dma-cells = < 0x4 >;
+			reg = < 0x40026400 0x400 >;
+			interrupts = < 0x38 0x0 0x39 0x0 0x3a 0x0 0x3b 0x0 0x3c 0x0 0x44 0x0 0x45 0x0 0x46 0x0 >;
+			clocks = < &rcc 0x0 0x400000 >;
+			st,mem2mem;
+			status = "disabled";
+			label = "DMA_2";
+		};
+		rng: rng@50060800 {
+			compatible = "st,stm32-rng";
+			reg = < 0x50060800 0x400 >;
+			clocks = < &rcc 0x1 0x40 >;
+			status = "okay";
+			label = "RNG";
+		};
+		i2c4: i2c@40006000 {
+			compatible = "st,stm32-i2c-v2";
+			clock-frequency = < 0x186a0 >;
+			#address-cells = < 0x1 >;
+			#size-cells = < 0x0 >;
+			reg = < 0x40006000 0x400 >;
+			clocks = < &rcc 0x2 0x1000000 >;
+			interrupts = < 0x5f 0x0 >, < 0x60 0x0 >;
+			interrupt-names = "event", "error";
+			status = "disabled";
+			label = "I2C_4";
+		};
+		spi6: spi@40015400 {
+			compatible = "st,stm32-spi-fifo", "st,stm32-spi";
+			#address-cells = < 0x1 >;
+			#size-cells = < 0x0 >;
+			reg = < 0x40015400 0x400 >;
+			clocks = < &rcc 0x3 0x200000 >;
+			interrupts = < 0x56 0x5 >;
+			status = "disabled";
+			label = "SPI_6";
+		};
+	};
+	cpus {
+		#address-cells = < 0x1 >;
+		#size-cells = < 0x0 >;
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m7";
+			reg = < 0x0 >;
+			#address-cells = < 0x1 >;
+			#size-cells = < 0x1 >;
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv7m-mpu";
+				reg = < 0xe000ed90 0x40 >;
+				arm,num-mpu-regions = < 0x8 >;
+			};
+		};
+	};
+	otghs_fs_phy: otghs_fs_phy {
+		compatible = "usb-nop-xceiv";
+		#phy-cells = < 0x0 >;
+		label = "OTGHS_FS_PHY";
+		phandle = < 0x4 >;
+	};
+	otgfs_phy: otgfs_phy {
+		compatible = "usb-nop-xceiv";
+		#phy-cells = < 0x0 >;
+		label = "OTGFS_PHY";
+		phandle = < 0x3 >;
+	};
+	sram0: memory@20010000 {
+		compatible = "mmio-sram";
+		reg = < 0x20010000 0x40000 >;
+	};
+	dtcm: memory@20000000 {
+		compatible = "arm,dtcm";
+		reg = < 0x20000000 0x10000 >;
+	};
+	pins {
```

Comment:
@tannewt 

I like this kind of syntax for defining the pins...what do you think?

Reply:
I don't like DTS's use of symbols for many things. I'm totally ok with how we do it now because it's much clearer to me.